### PR TITLE
WT-8068 Improve __rollback_row_modify control flow

### PR DESCRIPTION
**Pull request by etienneptl**
`__rollback_row_modify` now calls `__wt_row_modify`

### CODE REVIEW

**Review by kommiharibabu — APPROVED** [review]
LGTM. Thanks.

File: src/txn/txn_rollback_to_stable.c [hunk 1]
```diff
@@ -766,14 +737,23 @@ __rollback_abort_ondisk_kv(WT_SESSION_IMPL *session, WT_REF *ref, WT_ROW *rip, u
...
+            key = row_key;
+        else {
+            /* Unpack a row key. */
+            WT_ERR(__wt_scr_alloc(session, 0, &key));
+            WT_ERR(__wt_row_leaf_key(session, page, rip, key, false));
+        }
+        WT_ERR(__rollback_row_modify(session, ref, upd, key));
+    } else
         WT_ERR(__rollback_col_modify(session, ref, upd, recno));
     return (0);
```
Comment (kommiharibabu) [inline src/txn/txn_rollback_to_stable.c:751]: The `key` allocated doesn't get freed leads to the scratch buffer leak in the success scenarios. Rearrange the code to free the allocated scratch buffer.

Comment (etienneptl) [reply src/txn/txn_rollback_to_stable.c]: Thanks for spotting it, done in a8462afaa7e7b934ef8f7c53ffc169c16be6bde8

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Pull request merged**